src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/ResponseBodyTests.cs

```diff
@@ -30,5 +30,31 @@ public async Task ResponseBodyTest_FlushedPipeAndThenUnflushedPipe_AutoFlushed()
         {
             Assert.Equal(20, (await _fixture.Client.GetByteArrayAsync($"/FlushedPipeAndThenUnflushedPipe")).Length);
         }
+
+        [ConditionalFact]
+        [RequiresNewHandler]
+        public async Task ResponseBodyTest_BodyCompletionNotBlockedByOnCompleted()
+        {
+            Assert.Equal("SlowOnCompleted", await _fixture.Client.GetStringAsync($"/SlowOnCompleted"));
+        }
+
+        [ConditionalFact]
+        [RequiresNewHandler]
+        public async Task ResponseBodyTest_GettingHttpContextFieldsWork()
+        {
+            Assert.Equal("SlowOnCompleted", await _fixture.Client.GetStringAsync($"/OnCompletedHttpContext"));
+            Assert.Equal("", await _fixture.Client.GetStringAsync($"/OnCompletedHttpContext_Completed"));
+        }
+
+        [ConditionalFact]
+        [RequiresNewHandler]
+        public async Task ResponseBodyTest_CompleteAsyncWorks()
+        {
+            var response = await _fixture.Client.GetAsync("/CompleteAsync");
+            Assert.True(response.IsSuccessStatusCode);
+            var response2 = await _fixture.Client.GetAsync("/CompleteAsync_Completed");
+            Assert.True(response2.IsSuccessStatusCode);
```

Comment:
Comment on what is going on here. Its not obvious to the reader (me 😋) without looking at the server impl

Applies to all the tests that have multiple calls that depend on each other